src/components/text-input/index.md.njk

```diff
@@ -21,9 +21,15 @@ Do not use the text input component if you need to let users enter longer answer
 
 ## How it works
 
-All text inputs must have visible labels; placeholder text is not an acceptable replacement for a label as it vanishes when users start typing.
+Text inputs should usually have visible labels. However, in some cases, it's okay to use a visually hidden label. For example, if you're progressively revealing a text input with radios or checkboxes, and the input would otherwise be the same as the parent label.
```

Comment:
I've got a note for both sentences here, to hopefully increase clarity🤞

In my view, our top priority for this guidance is ensuring that the default is to have visible labels for text fields. This prevents potentially serious accessibility and usability issues in the vast majority of implementation cases. So I think we need to keep the strong wording of the sentence, and soften the exceptions piece.

```suggestion
All text inputs must have labels, and in the majority of cases the label should be visible. Using a visible label helps everyone understand the purpose of a text field.
```

This works to keep the original intent strong, but also adds space for potential exceptions where a visually-hidden label may be shown to perform better.

Reply:
CC @edwardhorsford here's my proposed re-edit for the latest commit I mentioned! Hopefully this creates enough room for the visually-hidden label exception you mentioned, while maintaining the strength of the guidance for the majority of other use-cases.

Reply:
Yeah my only concern was the certainty of the statement 'all text inputs must' - which I think would be taken as a global rule.

Reply:
Thanks all for contributing, definitely a nuance worth documenting 👍🏻